src/pudl/io_managers.py

```diff
@@ -276,6 +276,13 @@ def _handle_pandas_output(self, context: OutputContext, df: pd.DataFrame):
         table_name = self._get_table_name(context)
 
         sa_table = self._get_sqlalchemy_table(table_name)
+        pkg = Package.from_resource_ids()
+        all_resources = [resource.name for resource in pkg.resources]
+        if table_name in all_resources:
+            res = pkg.get_resource(table_name)
+            df = res.enforce_schema(df)
+        else:
+            df = pudl.metadata.fields.apply_pudl_dtypes(df)
```

Comment:
These changes intersect the work/discussion happening in #2445. It feels duplicative pass in a sqlalchemy metadata object created from the Package class to then create a package class inside of the IO Manager again.

I think it makes the most sense to:
1. Create a package object with the desired etl groups and pass it into a SQLiteIOManager object. `self.pkg.enforce_schema()` can be called in `handle_output` and `load_input`.
2. Create a new metadata field or etl_group that allows us to filter out SQL view tables so we can track the datatypes but avoid creating schemas for them in the db.